shared/lib/sharing/publicize/actions.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	FETCH_PUBLICIZE_CONNECTIONS,
+	RECEIVE_PUBLICIZE_CONNECTIONS,
+	FAIL_PUBLICIZE_CONNECTIONS_REQUEST
+} from 'lib/redux-store/action-types';
+
+/**
+ * Triggers a network request to fetch Publicize connections for the specified
+ * site ID.
+ *
+ * @param  {Number}   siteId Site ID
+ * @return {Function}        Action thunk
+ */
+export function fetchConnections( siteId ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: FETCH_PUBLICIZE_CONNECTIONS,
+			siteId
+		} );
+
+		return new Promise( ( resolve ) => {
```

Comment:
> Perhaps it's best that I remove the promise for now.

It's actually a pretty good precedent to open, so don't feel obligated just because I brought it up. I'm glad we're thinking about SSR and using promises for them!